android/src/main/java/com/tailscale/ipn/ui/localapi/Client.kt

```diff
@@ -128,12 +126,10 @@ class Client(private val scope: CoroutineScope) {
   fun putTaildropFiles(
       context: Context,
       peerId: StableNodeID,
-      files: Collection<FileTransfer>,
+      files: Collection<Ipn.OutgoingFile>,
       responseHandler: (Result<String>) -> Unit
   ) {
-    val manifest =
-        Json.encodeToString(
-            files.map { URLEncoder.encode(it.filename, "utf-8") to it.size }.toMap())
+    val manifest = Json.encodeToString(files)
```

Comment:
Manifest now just encodes the OutgoingFile objects. This lets us transfer the ID and actually simplifies the code on the Go side too.